src/sagemaker/predictor.py

```diff
@@ -196,11 +239,36 @@ def _create_request_args(
         if inference_id:
             args["InferenceId"] = inference_id
 
-        data = self.serializer.serialize(data)
+        if self.predictor_type == "async":
+            args["InputLocation"] = data
+            args.pop("ContentType", None)
+        else:
+            data = self.serializer.serialize(data)
+            args["Body"] = data
 
-        args["Body"] = data
         return args
 
+    def _wait_for_output(
+        self,
+        output_location,
+        check_period_in_second=5,
+    ):
+        """Placeholder docstring"""
+        output_url = urllib.parse.urlparse(output_location)
+        bucket = output_url.netloc
+        key = output_url.path[1:]
+
+        with stopit.ThreadingTimeout(10 * 60, swallow_exc=False):
+            while True:
+                try:
+                    return self.sagemaker_session.read_s3_file(bucket=bucket, key_prefix=key)
+                except ClientError as e:
+                    if e.response["Error"]["Code"] == "NoSuchKey":
+                        print("waiting for output...")
+                        time.sleep(check_period_in_second)
+                        continue
+                    raise
+
     def update_endpoint(
```

Comment:
Are we also going to support `update_endpoint` with async config and serverless config as well?